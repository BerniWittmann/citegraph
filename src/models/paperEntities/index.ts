import Author from './author'
import Conference from './conference'
import Country from './country'
import Institution from './institution'
import Publication from './publication'
import Publisher from './publisher'
import Record from './record'
import PaperEntity, { PaperEntityTableColumn, PaperEntityFields } from './base'

export const entities = [
  Record,
  Author,
  Institution,
  Country,
  Publication,
  Publisher,
  Conference
]

export const entityKeysMap = {
  [Record.key]: Record,
  [Author.key]: Author,
  [Institution.key]: Institution,
  [Country.key]: Country,
  [Publication.key]: Publication,
  [Publisher.key]: Publisher,
  [Conference.key]: Conference
}

export const entityKeys: Array<string> = entities.map((entity) => entity.key)

export type PaperEntityTableColumn = PaperEntityTableColumn

export type PaperEntity = PaperEntity

export type PaperEntityFields = PaperEntityFields

export default {
  Author,
  Conference,
  Country,
  Institution,
  Publication,
  Publisher,
  Record
}
