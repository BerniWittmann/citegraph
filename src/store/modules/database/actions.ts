import { ActionTree } from 'vuex'
import { RootState } from '@/store/types'
import { DatabaseState } from '@/store/modules/database/types'
import Axios from 'axios'

export const actions: ActionTree<DatabaseState, RootState> = {
  async clearDatabase ({ dispatch }): Promise<undefined> {
    try {
      await Axios(`/database/clear`, {
        method: 'POST'
      })
      dispatch('toasts/showSuccess', 'settings.database.clear.success', { root: true })
      return undefined
    } catch (error) {
      console.error(error)
      dispatch('toasts/showError', 'settings.database.clear.error', { root: true })
      return error
    }
  },

  async reseedDatabase ({ dispatch }): Promise<undefined> {
    try {
      await Axios(`/database/reseed`, {
        method: 'POST'
      })
      dispatch('toasts/showSuccess', 'settings.database.reseed.success', { root: true })
      return undefined
    } catch (error) {
      console.error(error)
      dispatch('toasts/showError', 'settings.database.reseed.error', { root: true })
      return error
    }
  }
}
