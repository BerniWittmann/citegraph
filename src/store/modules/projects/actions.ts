import Axios from 'axios'
import { ActionTree } from 'vuex'
import { ProjectsState } from '@/store/modules/projects/types'
import { RootState } from '@/store/types'
import Project from '@/models/project'
import * as types from './mutation-types'

export const actions: ActionTree<ProjectsState, RootState> = {
  async fetchProject ({ commit }, id: string): Promise<undefined> {
    try {
      const response = await Axios(`/projects/${id}`)
      const payload: Project = new Project(response && response.data)
      commit(types.SET_ACTIVE_PROJECT, payload)
      commit(types.OPEN_PROJECT, payload)
      return undefined
    } catch (error) {
      console.error(error)
      commit(types.UNSET_ACTIVE_PROJECT)
      return error
    }
  },

  unsetActiveProject ({ commit }): void {
    commit(types.UNSET_ACTIVE_PROJECT)
  },

  async fetchProjects ({ commit }): Promise<undefined> {
    try {
      const response = await Axios(`/projects`)
      const projects = response.data.map((project: Project) => new Project(project))
      commit(types.SET_PROJECTS, projects)
      return undefined
    } catch (error) {
      console.error(error)
      return error
    }
  },

  openProject ({ commit }, project: Project): void {
    commit(types.OPEN_PROJECT, project)
  },

  closeProject ({ commit }, project: Project): void {
    commit(types.CLOSE_PROJECT, project)
  },

  async createProject ({ commit }, data: Project): Promise<Project | undefined> {
    try {
      const response = await Axios(`/projects`, {
        method: 'POST',
        data: data
      })
      const project = new Project(response.data)
      commit(types.ADD_PROJECT, project)
      return project
    } catch (error) {
      console.error(error)
      return error
    }
  }
}
