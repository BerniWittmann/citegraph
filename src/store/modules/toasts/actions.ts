import { ActionTree } from 'vuex'
import { RootState } from '@/store/types'
import { ToastsState } from '@/store/modules/toasts/types'
import * as types from '@/store/modules/toasts/mutation-types'
import Toast, { ErrorToast, InfoToast, SuccessToast, WarningToast } from '@/models/toast'

export const actions: ActionTree<ToastsState, RootState> = {
  showSuccess ({ commit }, message: string): void {
    commit(types.ADD_TOAST, new SuccessToast(message))
  },

  showInfo ({ commit }, message: string): void {
    commit(types.ADD_TOAST, new InfoToast(message))
  },

  showWarning ({ commit }, message: string): void {
    commit(types.ADD_TOAST, new WarningToast(message))
  },

  showError ({ commit }, message: string): void {
    commit(types.ADD_TOAST, new ErrorToast(message))
  },

  removeToast ({ commit }, toast: Toast): void {
    commit(types.REMOVE_TOAST, toast)
  }
}
