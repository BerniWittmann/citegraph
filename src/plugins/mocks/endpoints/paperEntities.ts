import db from '../db'
import MockAdapter from 'axios-mock-adapter'
import { RecordFields } from '@/models/paperEntities/record'
import { setupMocks as setupGraphQLMocks } from './graphql-helpers/endpoint'
import { QueryConfiguration } from './graphql-helpers/queries'
import { AuthorFields } from '@/models/paperEntities/author'
import { CountryFields } from '@/models/paperEntities/country'

const records = db.getCollection('records')
const authors = db.getCollection('authors')
const countries = db.getCollection('countries')

const queryResponseConfiguration: Record<string, QueryConfiguration> = {
  'Records': {
    schemaName: 'record',
    queryName: 'Records',
    collection: records,
    filterFunction: recordsFilter
  },
  'Authors': {
    schemaName: 'author',
    queryName: 'Authors',
    collection: authors,
    filterFunction: authorsFilter
  },
  'Countries': {
    schemaName: 'country',
    queryName: 'Countries',
    collection: countries,
    filterFunction: countriesFilter
  }
}

function recordsFilter (obj: RecordFields, filter: string) {
  return obj.title.toLowerCase().includes(filter) ||
    obj.year.toString().includes(filter) ||
    obj.keywords.map(keyword => keyword.toLowerCase()).join('').includes(filter) ||
    obj.authors.map(author => (author.firstName + ' ' + author.lastName).toLowerCase()).join(';').includes(filter)
}

function authorsFilter (obj: AuthorFields, filter: string) {
  return obj.firstName.toLowerCase().includes(filter) ||
    obj.lastName.toLowerCase().includes(filter)
}

function countriesFilter (obj: CountryFields, filter: string) {
  return obj.name.toLowerCase().includes(filter)
}

export function setupMocks (mock: MockAdapter) {
  setupGraphQLMocks(mock, queryResponseConfiguration)
}
