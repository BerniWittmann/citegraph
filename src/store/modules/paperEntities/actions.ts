import { ActionTree } from 'vuex'
import { RootState } from '@/store/types'
import { PaperEntitiesState } from './types'
import { PaperEntityQueryParameters } from '@/models/paperEntities/query'
import Axios from 'axios'
import * as types from '@/store/modules/paperEntities/mutation-types'
import jsLogger from 'js-logger'
import { entityKeysMap, PaperEntity, PaperEntityFields } from '@/models/paperEntities'
import { PaperEntityMutationParameters } from '@/models/paperEntities/mutation'

const logger = jsLogger.get('paperEntities/actions')

export const actions: ActionTree<PaperEntitiesState, RootState> = {
  async fetchEntities ({ commit, dispatch }, params: PaperEntityQueryParameters): Promise<undefined> {
    try {
      const EntityClass = entityKeysMap[params.entityType]
      const response = await Axios(`/projects/${params.projectId}/paper-entities`, {
        method: 'POST',
        data: {
          query: EntityClass.getQuery(params)
        }
      })
      const data = response.data.data[EntityClass.queryName]
      const totalCount = data.count
      const payload: Array<PaperEntity> = EntityClass.transformer.fetchCollection(data[EntityClass.schemaName])
      commit(types.SET_ENTITIES, payload)
      commit(types.SET_ENTITY_COUNT, totalCount)
      commit(types.SET_ENTITY_TYPE, EntityClass.key)
      return undefined
    } catch (error) {
      logger.error(error)
      dispatch('toasts/showError', 'project.explore.fetch_error', { root: true })
      commit(types.SET_ENTITIES, [])
      commit(types.SET_ENTITY_COUNT, -1)
      commit(types.SET_ENTITY_TYPE, undefined)
      return error
    }
  },

  async fetchEntity ({ commit, dispatch }, params: PaperEntityQueryParameters): Promise<void> {
    try {
      const EntityClass = entityKeysMap[params.entityType]
      const response = await Axios(`/projects/${params.projectId}/paper-entities`, {
        method: 'POST',
        data: {
          query: EntityClass.getQuery(params)
        }
      })
      const data = response.data.data[EntityClass.queryName]
      const payload: PaperEntity = EntityClass.transformer.fetch(data[EntityClass.schemaName])
      commit(types.SET_ACTIVE_ENTITY, payload)
      return undefined
    } catch (error) {
      logger.error(error)
      dispatch('toasts/showError', 'project.explore.fetch_single_error', { root: true })
      commit(types.SET_ACTIVE_ENTITY, undefined)
      return error
    }
  },

  async updateEntity ({ commit, dispatch }, {
    params,
    data
  }: { params: PaperEntityMutationParameters, data: Partial<PaperEntityFields>}): Promise<void> {
    try {
      const EntityClass = entityKeysMap[params.entityType]
      const response = await Axios(`/projects/${params.projectId}/paper-entities`, {
        method: 'POST',
        data: {
          mutation: EntityClass.getMutation(params, EntityClass.transformer.send(data))
        }
      })
      const responseData = response.data.data[EntityClass.queryName]
      const payload: PaperEntity = EntityClass.transformer.fetch(responseData)
      commit(types.UPDATE_ENTITY, payload)
      return undefined
    } catch (error) {
      logger.error(error)
      dispatch('toasts/showError', 'project.explore.update_single_error', { root: true })
      return error
    }
  }
}
