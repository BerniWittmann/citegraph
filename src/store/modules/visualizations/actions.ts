import Axios from 'axios'
import { ActionTree } from 'vuex'
import { RootState } from '@/store/types'
import Visualization from '@/models/visualizations/Visualization'
import * as types from './mutation-types'
import jsLogger from 'js-logger'
import { VisalizationFields, visualizationsKeyMap } from '@/models/visualizations'
import { VisualizationsState } from '@/store/modules/visualizations/types'
const logger = jsLogger.get('visualizations/actions')

type VisualizationResponseFields = VisalizationFields & {
  key: string
}

function parseVisualization (visualization: VisualizationResponseFields): Visualization | undefined {
  if (!visualization || !visualization.key) return undefined
  const VisualizationClass = visualizationsKeyMap[visualization.key]
  if (!VisualizationClass) return undefined
  return VisualizationClass.transformer.fetch(visualization)
}

export const actions: ActionTree<VisualizationsState, RootState> = {
  async fetchVisualizations ({ commit, dispatch }, projectId: number): Promise<undefined> {
    try {
      const response = await Axios(`/projects/${projectId}/visualizations`)
      const visualizations: Array<Visualization> = response.data.map(parseVisualization).filter((v: Visualization | undefined) => !!v)
      commit(types.SET_VISUALIZATIONS, visualizations)
      return undefined
    } catch (error) {
      logger.error(error)
      dispatch('toasts/showError', 'visualizations.fetch_error', { root: true })
      return error
    }
  },

  async fetchVisualization ({ commit, dispatch }, { projectId, visualizationId }: { projectId: number, visualizationId: string }): Promise<void> {
    try {
      const response = await Axios(`/projects/${projectId}/visualizations/${visualizationId}`)
      const visualization: Visualization | undefined = parseVisualization(response.data)
      if (!visualization) throw new Error('Could not parse visualization')
      commit(types.SET_CURRENT_VISUALIZATION, visualization)
    } catch (error) {
      logger.error(error)
      dispatch('toasts/showError', 'visualizations.fetch_single_error', { root: true })
      return error
    }
  },

  unsetCurrentVisualization ({ commit }): void {
    commit(types.SET_CURRENT_VISUALIZATION, undefined)
  },

  async createVisualization ({ commit, dispatch }, { projectId, visualizationData, visualizationClass }: { projectId: number, visualizationData: Visualization, visualizationClass: typeof Visualization }): Promise<Visualization | undefined> {
    try {
      const response = await Axios(`/projects/${projectId}/visualizations`, {
        method: 'POST',
        data: {
          ...visualizationClass.transformer.send(visualizationData),
          data: undefined
        }
      })
      const visualization: Visualization | undefined = parseVisualization(response.data)
      if (!visualization) throw new Error('Could not parse visualization')
      commit(types.ADD_VISUALIZATION, visualization)
      dispatch('toasts/showSuccess', 'visualizations.add.successful', { root: true })
      return visualization
    } catch (error) {
      logger.error(error)
      dispatch('toasts/showError', 'visualizations.add.error', { root: true })
      return error
    }
  },

  async updateVisualization ({ commit, dispatch }, { projectId, visualizationData, visualizationClass }: { projectId: number, visualizationData: Visualization, visualizationClass: typeof Visualization }): Promise<Visualization | undefined> {
    try {
      const response = await Axios(`/projects/${projectId}/visualizations/${visualizationData.id}`, {
        method: 'PUT',
        data: {
          ...visualizationClass.transformer.send(visualizationData),
          data: undefined
        }
      })
      const visualization: Visualization | undefined = parseVisualization(response.data)
      if (!visualization) throw new Error('Could not parse visualization')
      commit(types.UPDATE_VISUALIZATION, visualization)
      dispatch('toasts/showSuccess', 'visualization.edit.successful', { root: true })
      return visualization
    } catch (error) {
      logger.error(error)
      dispatch('toasts/showError', 'visualization.edit.error', { root: true })
      return error
    }
  }
}
