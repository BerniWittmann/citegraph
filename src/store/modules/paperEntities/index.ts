import { Module } from 'vuex'
import { PaperEntitiesState } from './types'
import { RootState } from '@/store/types'
import { state } from './state'
import { actions } from './actions'
import { getters } from './getters'
import { mutations } from './mutations'

const namespaced: boolean = true

export const paperEntities: Module<PaperEntitiesState, RootState> = {
  namespaced,
  state,
  getters,
  actions,
  mutations
}
