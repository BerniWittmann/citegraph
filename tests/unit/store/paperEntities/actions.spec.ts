import moxios from 'moxios'
import { actions } from '@/store/modules/paperEntities/actions'
import { entityKeys, entityKeysMap } from '@/models/paperEntities'

function createDataWithId (id: number) {
  return {
    id: id.toString(),
    title: 'My title',
    authors: [{ id: '12', firstName: 'Author', lastName: 'LastName', countRecords: 12 }],
    keywords: ['my', 'keywords'],
    year: 2017,
    numberCitations: 12,
    firstName: 'Hans',
    lastName: 'Maier',
    countRecords: 12,
    name: 'Institution or CountryName'
  }
}

describe('store/modules/paperEntities/actions', () => {
  it('provides the actions', () => {
    expect(actions).toMatchSnapshot()
  })

  describe('fetchEntities', () => {
    const projectId = 42
    beforeEach(() => {
      moxios.install()
    })
    afterEach(() => {
      moxios.uninstall()
    })
    entityKeys.forEach((entityKey) => {
      describe('can fetch the entity type ' + entityKey, () => {
        const EntityClass = entityKeysMap[entityKey]
        const data = [
          new EntityClass(createDataWithId(1)),
          new EntityClass(createDataWithId(12))
        ]

        it('fetches the entities', (done) => {
          moxios.stubRequest(`/projects/${projectId}/paper-entities`, {
            status: 200,
            response: {
              data: {
                [EntityClass.queryName]: {
                  count: 23,
                  [EntityClass.schemaName]: data
                }
              }
            }
          })

          const onFulfilled = jest.fn()
          const commit = jest.fn()
          const dispatch = jest.fn()
          const action = actions.fetchEntities as Function

          action({ commit, dispatch }, {
            projectId,
            entityType: entityKey,
            perPage: 10,
            pageOffset: 1,
            filter: 'my filter',
            sortBy: 'title_ASC'
          }).then(onFulfilled)

          moxios.wait(() => {
            expect(onFulfilled).toHaveBeenCalledWith(undefined)
            // @ts-ignore
            const request = moxios.requests.mostRecent()
            expect(request.config.data).toMatchSnapshot()
            expect(commit).toHaveBeenCalledWith('SET_ENTITIES', data)
            expect(commit).toHaveBeenCalledWith('SET_ENTITY_TYPE', entityKey)
            expect(commit).toHaveBeenCalledWith('SET_ENTITY_COUNT', 23)
            done()
          })
        })

        it('handles the error when one appeared during the fetching', (done) => {
          moxios.stubRequest(`/projects/${projectId}/paper-entities`, {
            status: 400,
            response: 'The given query is invalid'
          })

          const onFulfilled = jest.fn()
          const commit = jest.fn()
          const dispatch = jest.fn()
          const action = actions.fetchEntities as Function

          action({ commit, dispatch }, {
            projectId,
            entityType: entityKey,
            perPage: 10,
            pageOffset: 1,
            filter: 'my filter',
            sortBy: 'title_ASC'
          }).then(onFulfilled)

          moxios.wait(() => {
            expect(onFulfilled).toHaveBeenCalledWith(new Error('Request failed with status code 400'))
            expect(dispatch).toHaveBeenCalledWith('toasts/showError', 'project.explore.fetch_error', { root: true })
            expect(commit).toHaveBeenCalledWith('SET_ENTITIES', [])
            expect(commit).toHaveBeenCalledWith('SET_ENTITY_TYPE', undefined)
            expect(commit).toHaveBeenCalledWith('SET_ENTITY_COUNT', -1)
            done()
          })
        })
      })
    })
  })
})
