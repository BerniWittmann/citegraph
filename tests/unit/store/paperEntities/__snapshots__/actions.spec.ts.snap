// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`store/modules/paperEntities/actions fetchEntities can fetch the entity type author fetches the entities 1`] = `"{\\"query\\":\\"{\\\\n      Authors(filter: my filter; orderBy: title_ASC; first: 10; skip: 10) {\\\\n        count\\\\n        author {\\\\n          id,\\\\n          first_name,\\\\n          last_name,\\\\n          number_citations\\\\n        }\\\\n      }\\\\n    }\\"}"`;

exports[`store/modules/paperEntities/actions fetchEntities can fetch the entity type conference fetches the entities 1`] = `"{\\"query\\":\\"{\\\\n      Conferences(filter: my filter; orderBy: title_ASC; first: 10; skip: 10) {\\\\n        count\\\\n        conference {\\\\n          name,\\\\n          number_citations\\\\n        }\\\\n      }\\\\n    }\\"}"`;

exports[`store/modules/paperEntities/actions fetchEntities can fetch the entity type country fetches the entities 1`] = `"{\\"query\\":\\"{\\\\n      Countries(filter: my filter; orderBy: title_ASC; first: 10; skip: 10) {\\\\n        count\\\\n        country {\\\\n          name,\\\\n          number_citations,\\\\n          flag_url\\\\n        }\\\\n      }\\\\n    }\\"}"`;

exports[`store/modules/paperEntities/actions fetchEntities can fetch the entity type institution fetches the entities 1`] = `"{\\"query\\":\\"{\\\\n      Institutions(filter: my filter; orderBy: title_ASC; first: 10; skip: 10) {\\\\n        count\\\\n        institution {\\\\n          name,\\\\n          number_citations\\\\n        }\\\\n      }\\\\n    }\\"}"`;

exports[`store/modules/paperEntities/actions fetchEntities can fetch the entity type publication fetches the entities 1`] = `"{\\"query\\":\\"{\\\\n      Publications(filter: my filter; orderBy: title_ASC; first: 10; skip: 10) {\\\\n        count\\\\n        publication {\\\\n          name,\\\\n          number_citations\\\\n        }\\\\n      }\\\\n    }\\"}"`;

exports[`store/modules/paperEntities/actions fetchEntities can fetch the entity type publisher fetches the entities 1`] = `"{\\"query\\":\\"{\\\\n      Publishers(filter: my filter; orderBy: title_ASC; first: 10; skip: 10) {\\\\n        count\\\\n        publisher {\\\\n          name,\\\\n          number_citations\\\\n        }\\\\n      }\\\\n    }\\"}"`;

exports[`store/modules/paperEntities/actions fetchEntities can fetch the entity type record fetches the entities 1`] = `"{\\"query\\":\\"{\\\\n      Records(filter: my filter; orderBy: title_ASC; first: 10; skip: 10) {\\\\n        count\\\\n        record {\\\\n          title,\\\\n          authors,\\\\n          year,\\\\n          keywords,\\\\n          number_citations\\\\n        }\\\\n      }\\\\n    }\\"}"`;

exports[`store/modules/paperEntities/actions fetchEntity can fetch the entity type author fetches the entity 1`] = `"{\\"query\\":\\"{\\\\n      Authors(id: 1) {\\\\n        count\\\\n        author {\\\\n          id,\\\\n          first_name,\\\\n          last_name,\\\\n          number_citations\\\\n        }\\\\n      }\\\\n    }\\"}"`;

exports[`store/modules/paperEntities/actions fetchEntity can fetch the entity type conference fetches the entity 1`] = `"{\\"query\\":\\"{\\\\n      Conferences(id: 1) {\\\\n        count\\\\n        conference {\\\\n          name,\\\\n          number_citations\\\\n        }\\\\n      }\\\\n    }\\"}"`;

exports[`store/modules/paperEntities/actions fetchEntity can fetch the entity type country fetches the entity 1`] = `"{\\"query\\":\\"{\\\\n      Countries(id: 1) {\\\\n        count\\\\n        country {\\\\n          name,\\\\n          number_citations,\\\\n          flag_url\\\\n        }\\\\n      }\\\\n    }\\"}"`;

exports[`store/modules/paperEntities/actions fetchEntity can fetch the entity type institution fetches the entity 1`] = `"{\\"query\\":\\"{\\\\n      Institutions(id: 1) {\\\\n        count\\\\n        institution {\\\\n          name,\\\\n          number_citations\\\\n        }\\\\n      }\\\\n    }\\"}"`;

exports[`store/modules/paperEntities/actions fetchEntity can fetch the entity type publication fetches the entity 1`] = `"{\\"query\\":\\"{\\\\n      Publications(id: 1) {\\\\n        count\\\\n        publication {\\\\n          name,\\\\n          number_citations\\\\n        }\\\\n      }\\\\n    }\\"}"`;

exports[`store/modules/paperEntities/actions fetchEntity can fetch the entity type publisher fetches the entity 1`] = `"{\\"query\\":\\"{\\\\n      Publishers(id: 1) {\\\\n        count\\\\n        publisher {\\\\n          name,\\\\n          number_citations\\\\n        }\\\\n      }\\\\n    }\\"}"`;

exports[`store/modules/paperEntities/actions fetchEntity can fetch the entity type record fetches the entity 1`] = `"{\\"query\\":\\"{\\\\n      Records(id: 1) {\\\\n        count\\\\n        record {\\\\n          title,\\\\n          authors,\\\\n          year,\\\\n          keywords,\\\\n          number_citations\\\\n        }\\\\n      }\\\\n    }\\"}"`;

exports[`store/modules/paperEntities/actions provides the actions 1`] = `
Object {
  "fetchEntities": [Function],
  "fetchEntity": [Function],
  "updateEntity": [Function],
}
`;

exports[`store/modules/paperEntities/actions updateEntity can update the entity type author updates the entity 1`] = `"{\\"mutation\\":\\"{\\\\n      Authors(id: 1) {\\\\n        author {\\\\n          first_name: \\\\\\"Hans\\\\\\", \\\\nlast_name: \\\\\\"Maier\\\\\\"\\\\n        }\\\\n      }\\\\n    }\\"}"`;

exports[`store/modules/paperEntities/actions updateEntity can update the entity type conference updates the entity 1`] = `"{\\"mutation\\":\\"{\\\\n      Conferences(id: 1) {\\\\n        conference {\\\\n          \\\\n        }\\\\n      }\\\\n    }\\"}"`;

exports[`store/modules/paperEntities/actions updateEntity can update the entity type country updates the entity 1`] = `"{\\"mutation\\":\\"{\\\\n      Countries(id: 1) {\\\\n        country {\\\\n          name: \\\\\\"Institution or CountryName\\\\\\", \\\\nflag_url: \\\\\\"undefined\\\\\\"\\\\n        }\\\\n      }\\\\n    }\\"}"`;

exports[`store/modules/paperEntities/actions updateEntity can update the entity type institution updates the entity 1`] = `"{\\"mutation\\":\\"{\\\\n      Institutions(id: 1) {\\\\n        institution {\\\\n          \\\\n        }\\\\n      }\\\\n    }\\"}"`;

exports[`store/modules/paperEntities/actions updateEntity can update the entity type publication updates the entity 1`] = `"{\\"mutation\\":\\"{\\\\n      Publications(id: 1) {\\\\n        publication {\\\\n          \\\\n        }\\\\n      }\\\\n    }\\"}"`;

exports[`store/modules/paperEntities/actions updateEntity can update the entity type publisher updates the entity 1`] = `"{\\"mutation\\":\\"{\\\\n      Publishers(id: 1) {\\\\n        publisher {\\\\n          \\\\n        }\\\\n      }\\\\n    }\\"}"`;

exports[`store/modules/paperEntities/actions updateEntity can update the entity type record updates the entity 1`] = `"{\\"mutation\\":\\"{\\\\n      Records(id: 1) {\\\\n        record {\\\\n          title: \\\\\\"My title\\\\\\"\\\\n        }\\\\n      }\\\\n    }\\"}"`;
